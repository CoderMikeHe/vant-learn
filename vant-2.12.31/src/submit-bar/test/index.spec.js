import SubmitBar from '..';
import { mount } from '../../../test';

test('submit event', () => {
  const submit = jest.fn();
  const wrapper = mount(SubmitBar, {
    context: {
      props: {
        price: 1,
      },
      on: { submit },
    },
  });

  const button = wrapper.find('.van-button');
  button.trigger('click');
  expect(submit).toHaveBeenCalled();
});

test('disable submit', () => {
  const submit = jest.fn();
  const wrapper = mount(SubmitBar, {
    context: {
      props: {
        price: 1,
        disabled: true,
      },
      on: { submit },
    },
  });

  expect(wrapper).toMatchSnapshot();

  // disabled
  const button = wrapper.find('.van-button');
  button.trigger('click');
  expect(submit).toHaveBeenCalledTimes(0);
});

test('without price', () => {
  const wrapper = mount(SubmitBar, {
    context: {
      props: {
        label: 'Label',
      },
    },
  });

  expect(wrapper).toMatchSnapshot();
});

test('top slot', () => {
  const wrapper = mount(SubmitBar, {
    scopedSlots: {
      top: () => 'Custom Top',
    },
  });

  expect(wrapper).toMatchSnapshot();
});

test('decimal-length prop', () => {
  const wrapper = mount(SubmitBar, {
    context: {
      props: {
        price: 111,
        decimalLength: 1,
      },
    },
  });

  expect(wrapper).toMatchSnapshot();
});

test('suffix-label prop', () => {
  const wrapper = mount(SubmitBar, {
    context: {
      props: {
        price: 111,
        label: 'Label',
        suffixLabel: 'Suffix Label',
      },
    },
  });

  expect(wrapper).toMatchSnapshot();
});

test('text-align prop', () => {
  const wrapper = mount(SubmitBar, {
    context: {
      props: {
        price: 111,
        textAlign: 'left',
      },
    },
  });
  expect(wrapper).toMatchSnapshot();
});

test('disable safe-area-inset-bottom prop', () => {
  const wrapper = mount(SubmitBar, {
    context: {
      props: {
        safeAreaInsetBottom: false,
      },
    },
  });
  expect(wrapper).toMatchSnapshot();
});

test('button-color prop', () => {
  const wrapper = mount(SubmitBar, {
    context: {
      props: {
        buttonColor: 'red',
      },
    },
  });
  expect(wrapper).toMatchSnapshot();
});

test('button slot', () => {
  const wrapper = mount(SubmitBar, {
    buttonText: 'text',
    scopedSlots: {
      button: () => 'Custom button',
    },
  });

  expect(wrapper).toMatchSnapshot();
});
