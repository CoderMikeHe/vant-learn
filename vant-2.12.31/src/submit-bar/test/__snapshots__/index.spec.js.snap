// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button slot 1`] = `
<div class="van-submit-bar">
  <div class="van-submit-bar__bar">Custom button</div>
</div>
`;

exports[`button-color prop 1`] = `
<div class="van-submit-bar">
  <div class="van-submit-bar__bar"><button class="van-button van-button--danger van-button--normal van-button--round van-submit-bar__button van-submit-bar__button--danger" style="color: white; background: red; border-color: red;">
      <div class="van-button__content"></div>
    </button></div>
</div>
`;

exports[`decimal-length prop 1`] = `
<div class="van-submit-bar">
  <div class="van-submit-bar__bar">
    <div class="van-submit-bar__text"><span>合计：</span><span class="van-submit-bar__price">¥<span class="van-submit-bar__price van-submit-bar__price--integer">1</span>.1</span></div><button class="van-button van-button--danger van-button--normal van-button--round van-submit-bar__button van-submit-bar__button--danger">
      <div class="van-button__content"></div>
    </button>
  </div>
</div>
`;

exports[`disable safe-area-inset-bottom prop 1`] = `
<div class="van-submit-bar van-submit-bar--unfit">
  <div class="van-submit-bar__bar"><button class="van-button van-button--danger van-button--normal van-button--round van-submit-bar__button van-submit-bar__button--danger">
      <div class="van-button__content"></div>
    </button></div>
</div>
`;

exports[`disable submit 1`] = `
<div class="van-submit-bar">
  <div class="van-submit-bar__bar">
    <div class="van-submit-bar__text"><span>合计：</span><span class="van-submit-bar__price">¥<span class="van-submit-bar__price van-submit-bar__price--integer">0</span>.01</span></div><button disabled="disabled" class="van-button van-button--danger van-button--normal van-button--disabled van-button--round van-submit-bar__button van-submit-bar__button--danger">
      <div class="van-button__content"></div>
    </button>
  </div>
</div>
`;

exports[`suffix-label prop 1`] = `
<div class="van-submit-bar">
  <div class="van-submit-bar__bar">
    <div class="van-submit-bar__text"><span>Label</span><span class="van-submit-bar__price">¥<span class="van-submit-bar__price van-submit-bar__price--integer">1</span>.11</span><span class="van-submit-bar__suffix-label">Suffix Label</span></div><button class="van-button van-button--danger van-button--normal van-button--round van-submit-bar__button van-submit-bar__button--danger">
      <div class="van-button__content"></div>
    </button>
  </div>
</div>
`;

exports[`text-align prop 1`] = `
<div class="van-submit-bar">
  <div class="van-submit-bar__bar">
    <div class="van-submit-bar__text" style="text-align: left;"><span>合计：</span><span class="van-submit-bar__price">¥<span class="van-submit-bar__price van-submit-bar__price--integer">1</span>.11</span></div><button class="van-button van-button--danger van-button--normal van-button--round van-submit-bar__button van-submit-bar__button--danger">
      <div class="van-button__content"></div>
    </button>
  </div>
</div>
`;

exports[`top slot 1`] = `
<div class="van-submit-bar">Custom Top<div class="van-submit-bar__bar"><button class="van-button van-button--danger van-button--normal van-button--round van-submit-bar__button van-submit-bar__button--danger">
      <div class="van-button__content"></div>
    </button></div>
</div>
`;

exports[`without price 1`] = `
<div class="van-submit-bar">
  <div class="van-submit-bar__bar"><button class="van-button van-button--danger van-button--normal van-button--round van-submit-bar__button van-submit-bar__button--danger">
      <div class="van-button__content"></div>
    </button></div>
</div>
`;
